olive/cli/base.py

```diff
@@ -417,35 +424,32 @@ def save_output_model(config: Dict, output_model_dir: Union[str, Path]):
     This assumes a single accelerator workflow.
     """
     run_output_path = Path(config["output_dir"])
-    if not any(run_output_path.rglob("model_config.json")):
-        # there must be an run_output_path with at least one model_config.json
+    model_config_path = run_output_path / "model_config.json"
+    if not model_config_path.exists():
         print("Command failed. Please set the log_level to 1 for more detailed logs.")
         return
 
     output_model_dir = Path(output_model_dir).resolve()
 
-    # hardlink/copy the output model to the output_model_dir
-    hardlink_copy_dir(run_output_path, output_model_dir)
-
-    # need to update the local path in the model_config.json
-    # should the path be relative or absolute? relative makes it easy to move the output
-    # around but the path needs to be updated when the model config is used
-    for model_config_file in output_model_dir.rglob("model_config.json"):
-        with model_config_file.open("r") as f:
-            model_config = json.load(f)
-
-        all_resources = find_all_resources(model_config)
-        for resource_key, resource_path in all_resources.items():
-            resource_path_str = resource_path.get_path()
-            if resource_path_str.startswith(str(run_output_path)):
-                set_nested_dict_value(
-                    model_config,
-                    resource_key,
-                    resource_path_str.replace(str(run_output_path), str(output_model_dir)),
-                )
-
-        with model_config_file.open("w") as f:
-            json.dump(model_config, f, indent=4)
+    with model_config_path.open("r") as f:
+        model_config = json.load(f)
+
+    all_resources = find_all_resources(model_config)
+    for resource_key, resource_path in all_resources.items():
+        src_path = Path(resource_path.get_path()).resolve()
+        if src_path.is_dir():
+            hardlink_copy_dir(src_path, output_model_dir / src_path)
```

Comment:
I don't think this is correct. `output_model_dir / src_path = src_path` since `src_path` is a fully resolved named. you probably wanted `str_path.name` like in 448?

Reply:
Yeah true! I forgot to update this!

Reply:
please also check how the outputs for a graph capture for with `--use_ort_genai` and `--use_model_builder` look. the `additional_files` attributes might make this copy the additional files into the output directory, even though they refer to files already in the `model` subdir.

Reply:
Like I described in the previous comment, I think it will be easier to just make saving the footprint, etc opt-in in the workflow config. that way, even for cli, we can just use the final output dir and not need to do any of this copy and path update.

Reply:
My intention is copying additional files to output folder as they are also a part of the output model.

Reply:
I think the additional files get copied twice in this. Once is as part of the `model_path` resource copy into `model` subdir. Then they are again copied individually directly into the output_dir itself.

Reply:
Are all additional files stored in the model path? I kinda remember it can be everywhere depending on the pass who created it. Was this logic updated before?

Reply:
it is stored in model path normally. some passes like modelbuilder with metadata saves it in a different folder but that was because we weren't sure if should copy/hardlink the existing model files. But both the pass carry forward https://github.com/microsoft/Olive/blob/main/olive/passes/olive_pass.py#L274 and cache model save have always saved in model_path https://github.com/microsoft/Olive/blob/0db2d72b00a3aeff3f859ea693b63d69c6de08eb/olive/cache.py#L451 

Since the output of a workflow goes through the cache model save, the additional files are already in model_path resources regardless of what pass was there before.

Reply:
I added `no_artifacts` to the Engine config, so no other artifacts will be saved to the output path except for the model files.

Reply:
Thanks!

Reply:
Other than the CI failures, the changes look good to me

Reply:
The tests were fixed.